mcs/class/corlib/System.IO/FileInfo.cs

```diff
@@ -230,13 +230,17 @@ public void MoveTo (string destFileName)
 		{
 			if (destFileName == null)
 				throw new ArgumentNullException ("destFileName");
-			if (destFileName == Name || destFileName == FullName)
+			if (destFileName.Length == 0)
+				throw new ArgumentException ("An empty file name is not valid.", "destFileName");
+
+			var destFullPath = Path.GetFullPath (destFileName);
+			if (destFullPath == FullPath)
 				return;
 			if (!File.Exists (FullPath))
 				throw new FileNotFoundException ();
 
-			File.Move (FullPath, destFileName);
-			this.FullPath = Path.GetFullPath (destFileName);
+			File.Move (FullPath, destFullPath);
+			this.FullPath = destFullPath;
```

Comment:
We should update OriginalPath here as well, otherwise .ToString() will be wrong.

Or should we just `return new FileInfo (destFullPath)?` like in CopyTo?